src/daal4py.cpp

```diff
@@ -290,7 +291,44 @@ daal::data_management::NumericTablePtr make_nt(PyObject * obj)
                 SET_NPY_FEATURE(PyArray_DESCR(ary)->type, MAKENT_, throw std::invalid_argument("Found unsupported array type"));
 #undef MAKENT_
             } else {
-                ptr = _make_npynt(obj);
+		if(array_is_behaved_F(ary) && (PyArray_NDIM(ary) == 2)) {
+		    int _axes = 0;
+		    npy_intp i = 0;
+		    npy_intp N = PyArray_DIM(ary, 1); // number of columns
+		    npy_intp column_len = PyArray_DIM(ary, 0);
+		    int array_type = PyArray_TYPE(ary);
+		    /*
+		     * Input is 2D F-contiguous array: represent it as SOA numeric table
+		     */
+		    daal::data_management::SOANumericTable * soatbl = NULL;
+
+		    // iterate over columns
+		    PyArrayIterObject * it = (PyArrayIterObject *) PyArray_IterAllButAxis(obj, &_axes);
+		    if (it == NULL) {
+			Py_XDECREF(it);
+			throw std::runtime_error("Creating DAAL SOA table from F-contigous NumPy array failed: iterator could not be created");
+		    }
+
+		    soatbl = new daal::data_management::SOANumericTable(N, column_len);
+		    
+		    while (PyArray_ITER_NOTDONE(it)) {
+			PyArrayObject *slice = (PyArrayObject *) PyArray_SimpleNewFromData(1, &column_len, array_type, (void *)PyArray_ITER_DATA(it));
```

Comment:
Why do we need a `slice` temporary? It seems to be used for ref-counting only and the current code keeps a double-indirection. Can't we use `ary`  directly for `NumpyDeleter` (and of course keep the `Py_INCREF`)?

Reply:
I did this initially, but `NumpyDeleter` has an assertion checking that the pointer being deleted coincides with the data pointer of the array, which it would not.